test/system/src/test/java/io/pravega/test/system/ReadWriteAndScaleWithFailoverTest.java

```diff
@@ -0,0 +1,459 @@
+    }
+
+
+
+    private CompletableFuture<Void> startWriting(final EventStreamWriter<Long> writer) {
+        return CompletableFuture.runAsync(() -> {
+            while (!testState.stopWriteFlag.get()) {
+                try {
+                    long value = testState.eventData.incrementAndGet();
+                    Thread.sleep(100);
+                    log.debug("Event write count before write call {}", value);
+                    writer.writeEvent(String.valueOf(value), value);
+                    log.debug("Event write count before flush {}", value);
+                    writer.flush();
+                    testState.eventWriteCount.getAndIncrement();
+                    log.debug("Writing event {}", value);
+                } catch (Throwable e ) {
+                    if (e instanceof InterruptedException) {
+                        log.error("error in sleep: ", e);
+                        break;
+                    }
+                    if (e instanceof ConnectionFailedException) {
+                        log.error("Test exception in writing events: ", e);
+                        break;
+                    }
+                    log.error("Test exception in writing events: ", e);
+                        break;
+                    }
+                }
+        }, executorService);
+    }
+
+    private CompletableFuture<Void> startReading(final EventStreamReader<Long> reader) {
+        return CompletableFuture.runAsync(() -> {
+            log.info("Exit flag status {}", testState.stopReadFlag.get());
+            log.info("Read count {} and write count {}", testState.eventReadCount.get(), testState.eventWriteCount.get());
+            while (!(testState.stopReadFlag.get() && testState.eventReadCount.get() == testState.eventWriteCount.get())) {
+                log.info("Entering read loop");
+                // exit only if exitFlag is true  and read Count equals write count.
+                try {
+                    final Long longEvent = reader.readNextEvent(SECONDS.toMillis(60)).getEvent();
+                    log.debug("Reading event {}", longEvent);
+                    if (longEvent != null) {
+                        //update if event read is not null.
+                        testState.eventsReadFromPravega.add(longEvent);
+                        testState.eventReadCount.incrementAndGet();
+                        log.debug("Event read count {}", testState.eventReadCount.get());
+                    } else {
+                        log.debug("Read timeout");
+                    }
+                } catch (Throwable e ) {
+                    if (e instanceof ConnectionFailedException) {
+                        log.error("Test exception in writing events: ", e);
+                        continue;
+                    } else {
+                        log.error("Test exception in writing events: ", e);
```

Comment:
This doesn't actually fail the test.
Maybe there should be a CompletableFuture in testState that can be failed.